fog/ingest/server/src/ingest_service.rs

```diff
@@ -113,7 +119,26 @@ where
                     .num_blocks()
                     .map_err(|err| rpc_database_err(err, logger))?,
             )
-            .map_err(|err| rpc_precondition_error("activate", err, logger))?;
+            .map_err(|err| match err {
+                IngestServiceError::ServerNotIdle => {
+                    rpc_precondition_error("activate", err, logger)
+                }
+                IngestServiceError::Connection(_) => rpc_unavailable_error("activate", err, logger),
+                IngestServiceError::Backup(PeerBackupError::Connection(_)) => {
+                    rpc_unavailable_error("activate", err, logger)
+                }
+                IngestServiceError::Backup(PeerBackupError::CreatingNewIngressKey) => {
```

Comment:
Can also be added to the match case proposed above.

Reply:
I think it will be nice to use the `|` operator to group together the precondition_failed/unavailable/permission/internal errors.